<launch>
    <arg name="input_point_topic" default="/hesai/pandar"/>
    <arg name="base_frame" default="p2"/>
    <arg name="clipping_height" default="2.2" />  <!-- Remove Points above this height value (default 2.0 meters) -->
    <arg name="min_point_distance" default="1.3" />  <!-- Removes Points closer than this distance from the sensor origin (default 1.85 meters) -->
    <arg name="radial_divider_angle" default="0.5" />  <!-- Angle of each Radial division on the XY Plane (default 0.08 degrees)-->
    <arg name="concentric_divider_distance" default="1.0" />  <!-- Distance of each concentric division on the XY Plane (default 0.0 meters) -->
    <arg name="local_max_slope" default="45" />  <!-- Max Slope of the ground between Points (default 8 degrees) -->
    <arg name="general_max_slope" default="45" />  <!-- Max Slope of the ground in the entire PointCloud, used when reclassification occurs (default 5 degrees)-->
    <arg name="min_height_threshold" default="0.05"/>  <!-- Minimum height threshold between points (default 0.05 meters)-->
    <arg name="reclass_distance_threshold" default="0.5" />  <!-- Distance between points at which re classification will occur (default 0.2 meters)-->
    <arg name="no_ground_point_topic" default="/points_no_ground" />
    <arg name="ground_point_topic" default="/points_ground" />


    <node pkg="ray_ground_filter" type="ray_ground_filter" name="ray_ground_filter" output="screen">
        <param name="input_point_topic" value="$(arg input_point_topic)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="clipping_height" value="$(arg clipping_height)" />
        <param name="min_point_distance" value="$(arg min_point_distance)" />
        <param name="radial_divider_angle" value="$(arg radial_divider_angle)" />
        <param name="concentric_divider_distance" value="$(arg concentric_divider_distance)" />
        <param name="local_max_slope" value="$(arg local_max_slope)" />
        <param name="general_max_slope" value="$(arg general_max_slope)" />
        <param name="min_height_threshold" value="$(arg min_height_threshold)" />
        <param name="reclass_distance_threshold" value="$(arg reclass_distance_threshold)" />
        <param name="no_ground_point_topic" value="$(arg no_ground_point_topic)" />
        <param name="ground_point_topic" value="$(arg ground_point_topic)" />
    </node>


    <rosparam command="load" file="$(find lidar_mapping)/config/params.yaml" />

    <env name="ROSCONSOLE_CONFIG_FILE"  value="$(find lidar_mapping)/config/log.config"/>

    <!-- <node pkg="tf" type="static_transform_publisher" name="velodyne_to_map" args="0 0 1 0 0 0 map velodyne 100" /> -->

    <node pkg="lidar_mapping" type="lidar_mapping_node" name="lidar_mapping" output="screen"/>
      
    <node pkg="rviz" type="rviz" name="rviz" args="-d /home/wjq/workspace/smartcar_ws/src/mapping/lidar_mapping/config/rviz.rviz" output="screen" />

    <param name="/use_sim_time" value="false" />
</launch>
